Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -347,6 +348,54 @@ Value ArrayPrototype::index_of(Interpreter& interpreter)
     return Value(-1);
 }
 
+Value ArrayPrototype::reduce(Interpreter& interpreter)
+{
+    auto this_value = interpreter.this_value();
+    auto* this_object = this_value.to_object(interpreter);
+    if (!this_object)
+        return {};
+
+    auto initial_length = get_length(interpreter, *this_object);
+    if (interpreter.exception())
+        return {};
+
+    size_t start = 1;
+
+    auto accumulator = js_undefined();
+    if (interpreter.argument_count() > 1) {
+        accumulator = interpreter.argument(1);
+        start = 0;
+    } else if (initial_length >= 1) {
+        accumulator = this_object->get_by_index(0);
```

Comment:
Missing exception check:


```suggestion
        accumulator = this_object->get_by_index(0);
        if (interpreter.exception())
            return {};
```